packages/nimble-components/src/banner/specs/README.md

```diff
@@ -0,0 +1,186 @@
+
+In this initial implementation, we will not limit the height of the banner. It will grow to fit the text that it is given.
+
+When the user presses the close button, we will hide the banner (and fire a `close` event).
+
+### API
+
+*The key elements of the component's public API surface:*
+
+- *Component Name*: `nimble-banner`
+- *Props/Attrs*:
+  - `heading` - if set, a heading with the specified text is shown to the left of the message. OPEN QUESTION: should we instead use the existing `title` attribute?
+  - `text` - the message text
+  - `prevent-dismiss` - set to hide the close button (attr name taken from Nimble dialog)
+  - `type` - one of `error`, `warning`, `info`, or `neutral` (default)
+  - `action-text` - the text of the action link/button. If unset, no link/button is displayed.
+  - `action-href` - if set, the action element will be a link with the given `href`. (Otherwise, when `action-text` is set, a button element will be displayed.)
+  - `action-button-appearance` - either `outline` or `ghost` (default). When an action button is displayed, this controls the appearance variant.
+  - `close-button-title` - a localized title to give the close button (for a11y purposes)
+- *Methods*
+- *Events*
+  - `close` - fired when the banner is dismissed
+  - `action` - fired when the action button is activated (by mouse or keyboard). This is the intended way for clients to perform an action.
+- *CSS Classes and CSS Custom Properties that affect the component*
+
+### Alternatives
+
+- The text of the banner could be provided as content rather than an attribute. This has the benefit of leaving the door open for us to support more than just plaintext in the future. However, so long as we _don't_ want to support arbitrary HTML, it may have the downside of giving users the wrong idea.
+
+- Instead of configuring the action link/button via attributes, we could allow the user to provide that element as content and slot it in an `action` slot. This would have the following benefits:
+    - allows full configuration of action element (e.g. attributes like `target` on a `nimble-anchor`)
+    - user can add handlers directly to their action element rather than relying on `action` event
+    - easier to support multiple actions in the future
+    - easy to support other kinds of action elements in the future
+
+    However, that flexibility is also a downside. We would no longer have tight control over what the user could show and how it would look.
+
+### Anatomy
+
+```html
+<div class="icon">
+  ${when(x => x.type === 'error', html`<nimble-icon-exclamation-mark></nimble-icon-exclamation-mark>`)}
```

Comment:
nit: It's not important for this example, but when you get to actual implementation, these should use `DesignSystem.tagFor()` rather than hard-coding element names.